GitHubRepositories.ps1

```diff
@@ -461,6 +461,98 @@ function Get-GitHubRepository
     return Invoke-GHRestMethodMultipleResult @params
 }
 
+function Rename-GitHubRepository
+{
+<#
+    .SYNOPSIS
+        Rename a GitHub repository
+
+    .DESCRIPTION
+        Renames a GitHub repository with the new name provided.
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER Uri
```

Comment:
Was there a specific reason that you opted to to not allow the alternate `Elements` parameter set for this method?  It creates an inconsistency in use vs all of the other methods in the module.

I *love* that you did that `process` work so that things could be piped in (we need more of that in this module (see #63)), but I don't think it should be at the sacrifice of input consistency across the functions.

Reply:
Yes, understood.  And, sheesh:  my comment ended up being so long that I added a T.o.C. to combat/prevent the `TLDR;`:

- The specific reason for no `Elements` ParameterSet so far
- Example of the basis for the reasoning
- I agree:  consistency wins
- Wilco: I'll add that ParameterSet in the near future

#### The specific reason
I was using the established design of, "use Get-* cmdlets for getting, and pipe them to the objects' corresponding Set-/Rename-/Remove-/Update-* cmdlets".

#### Example
`Microsoft.PowerShell.Management\Get-Item` has lots of params for getting items, and `Microsoft.PowerShell.Management\Rename-Item` has just the provision of specifying an item's path (granted, via either `-Path` or `-LiteralPath` params)  That is, `Rename-Item` does not have the variety of "item getting parameters" that the `Get-Item` cmdlet does -- we would use `Get-Item` for all of that filter/include/exclude kind of activity.  I've recognized that practice is several modules, and tend to embrace it -- use `Get-ObjectX` to get the `objectX` thing, and then use that object as the input to all the other `*-ObjectX` cmdlets.

#### I agree
As you say, though, this module does not yet have wide support for `ValueFromPipeline` on cmdlets, and we do not yet have an object model by which to have a parameter like, say, `-Repository`.  I agree, it is probably best to keep input consistency across cmdlets, and maybe wait to consider my above approach until such a time that we/someone increases that `ValueFromPipeline` support for the module.

#### Wilco
I'll add the `Elements` ParameterSet to the `Rename-GitHubRepository` cmdlet to support `-OwnerName` and `-RepositoryName` params.  I expect to make time in the next few days to do so, to re-test, etc.

Reply:
Added in 109b977303396520c34b1e3b0566b47adbe3e998